btcstaking-tracker/stakingeventwatcher/stakingeventwatcher.go

```diff
@@ -454,6 +461,65 @@ func (sew *StakingEventWatcher) watchForSpend(spendEvent *notifier.SpendEvent, t
 	)
 }
 
+func (sew *StakingEventWatcher) buildSpendingTxProof(spendingTx *wire.MsgTx) (*btcstakingtypes.InclusionProof, error) {
+	txHash := spendingTx.TxHash()
+	if len(spendingTx.TxOut) == 0 {
+		return nil, fmt.Errorf("stake spending tx has no outputs")
```

Comment:
tbh this should not happen as we are handling tx after we detected it was included in mempool/chain. Maybe panic here or have some error which we do not retry ? (imo this means our assumptions are broken so we should do something drastic)